### DISCUSSION TITLE
Config file usage

### DISCUSSION THREAD

**Discussion by teamjoetv** [streamlink/streamlink#5223] [Q&A] [answered]
### Description

Hi,

Is there a way to add filename and file path in config parameter? I have been trying to add a config parameter which contains http-headers somethink like below. However the paths that I've been trying don't work at all

>> streamlink --config "../t_config.txt" --stdout "https://test.abc/live5000.smil/playlist.m3u8" best

>> streamlink --config "~/test1/t_config.txt" --stdout "https://test.abc/live5000.smil/playlist.m3u8" best

>> >> streamlink --config "/test1/test2/t_config.txt" --stdout "https://test.abc/live5000.smil/playlist.m3u8" best

**Comment by bastimeyer** [comment, marked as answer]
The `str` value of each `--config` parameter (multiple params are supported) gets turned into a `pathlib.Path` object, as you can see here:
https://github.com/streamlink/streamlink/blob/5.3.1/src/streamlink_cli/main.py#L649-L655

This means that relative paths are resolved automatically (as well as the `~` prefix for the user's home directory via the `expandhome()` call) when checking the existence of the config file before it gets loaded. Relative paths obviously depend on your current working directory when invoking the `streamlink` command.

Config resolving is fully tested:
https://github.com/streamlink/streamlink/blob/5.3.1/tests/cli/test_main_setup_config_args.py

Example with a relative path of a config which sets the `--version` parameter:
```sh
# no parameters (default)
$ streamlink
usage: streamlink [OPTIONS] <URL> [STREAM]

Use -h/--help to see the available options or read the manual at https://streamlink.github.io

# create custom config which sets the --version parameter and show the current working directory
$ echo "version" > /media/storage/streamlink-test-config
$ echo $PWD
/home/basti

# load the custom config from a relative and from an absolute path
$ streamlink --config ../../media/storage/streamlink-test-config 
streamlink 5.3.1+22.g3178ac3a
$ streamlink --config /media/storage/streamlink-test-config
streamlink 5.3.1+22.g3178ac3a
$ streamlink --config ~/../../media/storage/streamlink-test-config 
streamlink 5.3.1+22.g3178ac3a
```

Make sure that you have set the correct paths and that you are using the correct syntax in your config files:
https://streamlink.github.io/cli/config.html#syntax

`--loglevel=debug` always lets you know which parameters were set or loaded from any config file.
https://streamlink.github.io/cli.html#cmdoption-loglevel

**Reply by bastimeyer** [reply]
Btw, the `--http-header` CLI parameter requires its value to be in the format `header=value`, which means the config file should look like

```
http-header=foo=bar
http-header=bar=qux
```

https://streamlink.github.io/cli.html#cmdoption-http-header

**Reply by teamjoetv** [reply]
Sweet. The ways you mentioned are perfectly working. Really appreciated!
